src/gsi/read_iasing.f90

```diff
@@ -0,0 +1,950 @@
+
+  if (jsatid == 'metop-sg-a1') then
+     kidsat=24
+  else
+     write(*,*) 'READ_IASI-NG: Unrecognized value for jsatid '//jsatid//': RETURNING'
+     return
+  endif
+ 
+!  write(6,*)'READ_IASI-NG: mype, mype_root,mype_sub, npe_sub,mpi_comm_sub', &
+!          mype, mype_root,mype_sub,mpi_comm_sub
+
+  radedge_min = 0
+  radedge_max = 1000
+
+! Find the iasi-ng offset in the jpch_rad list.  This is for the iuse flag
+! and count the number of cahnnels in the satinfo file 
+  ioff=jpch_rad
+  subset_start = 0
+  subset_end = 0
+  assim = .false.
+  do i=1,jpch_rad
+     if (trim(nusis(i))==trim(sis)) then
+        ioff = min(ioff,i)   ! iasi-ng offset
+        if (subset_start == 0) then
+          if (radedge1(i)/=-1 .and. radedge2(i)/=-1) then
+             radedge_min=radedge1(i)
+             radedge_max=radedge2(i)
+          end if
+          subset_start = i
+        endif
+        if (iuse_rad(i) > 0) assim = .true.  ! Are any of the IASI-NG channels being used?
+        subset_end = i 
+     endif
+  end do 
+  satinfo_nchan = subset_end - subset_start + 1
+  allocate(channel_number(satinfo_nchan))
+  allocate(sc_index(satinfo_nchan))
+  allocate(bufr_index(satinfo_nchan)) 
+  ioff = ioff - 1 
+
+! If all channels of a given sensor are set to monitor or not
+! assimilate mode (iuse_rad<1), reset relative weight to zero.
+! We do not want such observations affecting the relative
+! weighting between observations within a given thinning group.
+  if (.not.assim) val_iasing=zero
+
+  if (mype_sub==mype_root) write(6,*)'READ_IASI-NG:  iasi-ng offset ',ioff
+
+  senname = 'IASI-NG'
+  
+  allspotlist= &
+   'SAID YEAR MNTH DAYS HOUR MINU SECO CLATH CLONH SAZA BEARAZ SOZA SOLAZI'
+
+! load spectral coefficient structure  
+  quiet=.not. verbose
+
+  imager_coeff = .false.
+  spc_filename =trim(crtm_coeffs_path)//'metimage_'//trim(jsatid)//'.SpcCoeff.bin'
+  inquire(file=trim(spc_filename), exist=imager_coeff)
+  if ( imager_coeff ) then
+    allocate( sensorlist(2))
+    sensorlist(1) = sis
+    sensorlist(2) = 'metimage_'//trim(jsatid)
+  else
+    allocate( sensorlist(1))
+    sensorlist(1) = sis
+  endif
+
+  if( crtm_coeffs_path /= "" ) then
+     if(mype_sub==mype_root .and. print_verbose) write(6,*)'READ_IASI-NG: crtm_spccoeff_load() on path "'//trim(crtm_coeffs_path)//'"'
+     error_status = crtm_spccoeff_load(sensorlist,&
+        File_Path = crtm_coeffs_path,quiet=quiet )
+  else
+     error_status = crtm_spccoeff_load(sensorlist,quiet=quiet)
+  endif
+
+  if (error_status /= success) then
+     write(6,*)'READ_IASI-NG:  ***ERROR*** crtm_spccoeff_load error_status=',error_status,&
+        '   TERMINATE PROGRAM EXECUTION'
+     call stop2(71)
+  endif
+
+!  find IASI-NG sensorindex
+  sensorindex_iasing = 0
+  if ( sc(1)%sensor_id(1:7) == 'iasi-ng' .or. sc(1)%sensor_id == '999') then
+!TODO This is the correct code once the CRTM spectral coefficient is fixed.          
+!TODO JAJ  if ( sc(1)%sensor_id(1:7) == 'iasi-ng' ) then   
```

Comment:
We can remove `JAJ`.  To make things clear to those who read this code, it would be good to mention in a comment that line 300 should be removed when CRTM spectral coefficient file is fixed.  That is, line 300 is a temporary code.  Line 302 is the line we activate after the fix file is corrected.

Reply:
Suggested change
```
!TODO The line below is a temporary fix to work around a problem in the CRTM spectral coefficient file
  if ( sc(1)%sensor_id(1:7) == 'iasi-ng' .or. sc(1)%sensor_id == '999') then
!TODO  Remove the above line and activate the line below once the CRTM spectral coefficient file is fixed
!TODO if ( sc(1)%sensor_id(1:7) == 'iasi-ng' ) then   
```

Reply:
Changed comments as requested